app/features/AccountSlice.ts

```diff
@@ -427,6 +455,35 @@ async function updateAccountFromAccountInfo(
         );
     }
 
+    let validatorId: bigint | undefined;
+    if (accountInfo.type === AccountInfoType.Baker) {
+        validatorId = accountInfo.accountBaker.bakerId;
+    } else if (
+        accountInfo.type === AccountInfoType.Delegator &&
+        accountInfo.accountDelegation.delegationTarget.delegateType ===
+            DelegationTargetType.Baker
+    ) {
+        validatorId = accountInfo.accountDelegation.delegationTarget.bakerId;
+    }
+
+    if (validatorId !== undefined) {
+        const poolStatus = await getPoolStatusLatest(validatorId);
+        if (poolStatus.isSuspended) {
+            dispatch(
+                setSuspensionStatus({
+                    address: account.address,
+                    status: 'suspended',
+                })
+            );
+        } else if (poolStatus.currentPaydayStatus?.isPrimedForSuspension) {
+            dispatch(
+                setSuspensionStatus({
+                    address: account.address,
+                    status: 'primed',
+                })
+            );
+        }
```

Comment:
Maybe not a big issue, but from my understanding the validator can technically be both suspended and primed for suspension at the same time.

Reply:
Yes, to my understanding, being flagged 'primed for suspension' is separate from the actual suspension state.
Meaning you could be 'primed for suspension' and then self-suspend and unsuspend again, resulting in still being primed for suspension.